src/elexmodel/handlers/data/CombinedData.py

```diff
@@ -132,48 +156,35 @@ def _get_unexpected_units(self, percent_reporting_threshold, turnout_factor_lowe
         expected_geographic_units = self._get_expected_geographic_unit_fips().tolist()
         no_baseline_units = self._get_units_without_baseline()
         # Note: this uses current_data because self.data drops unexpected units
-        unexpected_units = self.current_data[
-            ~self.current_data["geographic_unit_fips"].isin(expected_geographic_units)
-            | self.current_data.geographic_unit_fips.isin(no_baseline_units)
-        ].reset_index(drop=True)
-
-        units_with_strange_turnout_factor = self.data[
-            (self.data.percent_expected_vote >= percent_reporting_threshold)
-            & ((self.data.turnout_factor <= turnout_factor_lower) | (self.data.turnout_factor >= turnout_factor_upper))
-        ][self.current_data.columns]
-
-        all_unexpected_units = pd.concat([unexpected_units, units_with_strange_turnout_factor]).reset_index(drop=True)
-        all_unexpected_units.drop_duplicates(subset="geographic_unit_fips", inplace=True)
-        return all_unexpected_units
-
-    def get_unexpected_units(self, percent_reporting_threshold, aggregates, turnout_factor_lower, turnout_factor_upper):
-        """
-        Gets units for which we will not be making predictions:
-            - unexpected units (ie. units for which we have no covariates prepared)
-            - units for which the baseline results is zero (ie. units that are tiny)
-            - units with strange turnout factors (ie. units that are likely precinct mismatches)
-        """
-
-        unexpected_units = self._get_unexpected_units(
-            percent_reporting_threshold, turnout_factor_lower, turnout_factor_upper
+        unexpected_units = (
+            self.current_data[
+                ~self.current_data["geographic_unit_fips"].isin(expected_geographic_units)
+                | self.current_data.geographic_unit_fips.isin(no_baseline_units)
+            ]
+            .reset_index(drop=True)
+            .drop_duplicates(subset="geographic_unit_fips")
+            .copy()
         )
 
-        # since we were not expecting them, we have don't have their county or district
-        # from preprocessed data. so we have to add that back in.
-        if "county_fips" in aggregates:
-            unexpected_units["county_fips"] = unexpected_units["geographic_unit_fips"].apply(
-                self._get_county_fips_from_geographic_unit_fips
-            )
-
-        if "district" in aggregates:
-            unexpected_units["district"] = unexpected_units["geographic_unit_fips"].apply(
-                self._get_district_from_geographic_unit_fips
-            )
+        units_with_strange_turnout_factor = (
```

Comment:
I think it would make sense to split this into two functions. One that actually gets the unexpected units and one that gets the nonmodeled units (ie ones with strange turnout factors etc.)

Reply:
I can do that 🎉 but here's a question...this is probably insanely rare, but if there's a unit that is both unexpected _and_ non-predictive, which of the two buckets should the unit be placed in?  Or should it be included in both? 🤔

Reply:
That cannot happen. We can only measure whether a unit is non-predictive if we have a baseline result. If it's unexpected we have no preprocessed data, so it doesn't have a baseline result.

Reply:
Got it!  Ok, in that case I was able to split this into two functions, but I did have to add a couple rules to `_get_non_modeled_units()` so that only expected units and units with baselines could be considered as potential non-modeled units.  This fits with these rules and without this, we have two unit tests that would count the same units as unexpected and non-modeled 👀 🎉